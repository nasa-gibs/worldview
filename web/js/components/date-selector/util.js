import util from '../../util/util';

export const monthStringArray = [
  'JAN',
  'FEB',
  'MAR',
  'APR',
  'MAY',
  'JUN',
  'JUL',
  'AUG',
  'SEP',
  'OCT',
  'NOV',
  'DEC',
];

/**
 *
 * @param {*} value
 * @param {*} date
 * @returns
 */
export const yearValidation = (value, dateParam, validate) => {
  const date = new Date(dateParam);
  if (value > 1000 && value < 9999) {
    const newDate = new Date(date.setUTCFullYear(value));
    return validate(newDate);
  }
  return null;
};

/**
 *
 * @param {*} value
 * @param {*} date
 * @returns
 */
export const monthValidation = (value, dateParam, validate) => {
  const date = new Date(dateParam);
  let newDate;
  // eslint-disable-next-line no-restricted-globals
  if (!isNaN(value) && value < 13 && value > 0) {
    newDate = new Date(date.setUTCMonth(value - 1));
    if (newDate) {
      return validate(newDate);
    }
    return null;
  }
  const realMonth = util.stringInArray(monthStringArray, value);
  if (realMonth !== false) {
    const day = date.getUTCDate();
    const zeroDay = new Date(date.setUTCDate(1));

    const zeroAddMonth = new Date(zeroDay.setUTCMonth(realMonth));
    const zeroAddedMonthNumber = zeroAddMonth.getUTCMonth();

    const addDay = new Date(zeroAddMonth.setUTCDate(day));
    const addedDayMonthNumber = addDay.getUTCMonth();

    if (addedDayMonthNumber !== zeroAddedMonthNumber) {
      return false;
    }
    return validate(addDay);
  }
  return null;
};

/**
 *
 * @param {*} value
 * @param {*} date
 * @returns
 */
export const dayValidation = (value, dateParam, validate) => {
  const date = new Date(dateParam);
  const standardMaxDateForMonth = 31;

  if (value > 0 && value <= standardMaxDateForMonth) {
    const actualMaxDateForMonth = new Date(
      date.getYear(),
      date.getMonth() + 1,
      0,
    ).getDate();

    if (value > actualMaxDateForMonth) {
      return false;
    }
    const newDate = new Date(date.setUTCDate(value));
    return validate(newDate);
  }
  return null;
};

/**
 *
 * @param {*} value
 * @param {*} date
 * @returns
 */
export const hourValidation = (value, dateParam, validate) => {
  const date = new Date(dateParam);
  if (value >= 0 && value <= 23) {
    const newDate = new Date(date.setUTCHours(value));
    return validate(newDate);
  }
  return null;
};

/**
 *
 * @param {*} value
 * @param {*} date
 * @returns
 */
export const minuteValidation = (value, dateParam, validate) => {
  const date = new Date(dateParam);
  if (value >= 0 && value <= 59) {
    const newDate = new Date(date.setUTCMinutes(value));
    return validate(newDate);
  }
  return null;
};
