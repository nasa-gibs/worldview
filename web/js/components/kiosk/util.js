// formats date for kiosk mode and updates to EST
// eslint-disable-next-line import/prefer-default-export
export function formatKioskDate(date, subdaily) {
  const options = {
    year: 'numeric',
    month: 'long',
    day: '2-digit',
    hour: '2-digit',
    minute: '2-digit',
    hour12: false,
    timeZone: 'America/New_York',
  };

  const formatter = new Intl.DateTimeFormat('en-US', options);
  const dateParts = formatter.formatToParts(date);

  const year = dateParts.find((part) => part.type === 'year').value;
  const month = dateParts.find((part) => part.type === 'month').value.slice(0, 3);
  const day = dateParts.find((part) => part.type === 'day').value;
  const hours = dateParts.find((part) => part.type === 'hour').value;
  const minutes = dateParts.find((part) => part.type === 'minute').value;

  const formattedDate = subdaily ? `${day} ${month} ${year} ${hours}:${minutes}:00 EDT` : `${day} ${month} ${year}`;

  return formattedDate;
}


