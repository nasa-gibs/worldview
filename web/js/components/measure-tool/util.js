import {
  MultiLineString as OlGeomMultiLineString,
  Polygon as OlGeomPolygon,
} from 'ol/geom';
import geographiclib from 'geographiclib';

const geod = geographiclib.Geodesic.WGS84;
const geographicProj = 'EPSG:4326';

/**
 * Shift x value of every coord except the last,
 * which is the active drawing point
 * @param {*} coords
 * @param {*} shiftValue
 */
function shiftXCoords (coords, shiftValue) {
  const newCoordsArray = [];
  const len = coords.length;
  for (let i = 0; i < len - 1; i += 1) {
    const [x, y] = coords[i];
    newCoordsArray.push([x + shiftValue, y]);
  }
  newCoordsArray.push(coords[len - 1]);
  return newCoordsArray;
}

/**
 * If the last two coordinates in a measurement are more than 180
 * degrees apart, flip the whole drawing forward or backward 360 degrees
 * @param {*} geom
 */
function checkForXFlip(geom, projection) {
  const coords = geom.getCoordinates();
  const [x1] = coords[coords.length - 2];
  const [x2] = coords[coords.length - 1];
  if (Math.abs(x1 - x2) > 180 && projection === geographicProj) {
    if (x1 < x2) {
      geom.setCoordinates(shiftXCoords(coords, 360));
    } else if (x1 > x2) {
      geom.setCoordinates(shiftXCoords(coords, -360));
    }
  }
  return geom;
}

/**
 * Transforms a LineString of two points to a MultiLineString of multiple points
 * applying a great circle arc transformation
 * @param {*} geom - the geometry object to apply great circle arc transformation to
 */
export function transformLineStringArc(geom, projection) {
  const coords = [];
  const distance = 10000; // meters between segments
  const transformedGeom = checkForXFlip(geom, projection).clone().transform(projection, geographicProj);
  transformedGeom.forEachSegment((segStart, segEnd) => {
    const line = geod.InverseLine(segStart[1], segStart[0], segEnd[1], segEnd[0]);
    const n = Math.ceil(line.s13 / distance);
    for (let i = 0; i <= n; i += 1) {
      const s = Math.min(distance * i, line.s13);
      const r = line.Position(s, geographiclib.Geodesic.LONG_UNROLL);
      coords.push([r.lon2, r.lat2]);
    }
  });
  return new OlGeomMultiLineString([coords]).transform(geographicProj, projection);
}

/**
 * Transforms a Polygon into one with addiitonal points on each edge to account for
 * great circle arc
 * @param {*} geom - the geometry object to apply great circle arc transformation to
 */
export function transformPolygonArc(geom, projection) {
  const coords = [];
  const transformedGeom = geom.clone().transform(projection, geographicProj);
  const distance = 10000; // meters between segments
  const polyCoords = transformedGeom.getCoordinates()[0];
  for (let i = 0; i < polyCoords.length - 1; i += 1) {
    const line = geod.InverseLine(
      polyCoords[i][1],
      polyCoords[i][0],
      polyCoords[i + 1][1],
      polyCoords[i + 1][0],
    );
    const n = Math.ceil(line.s13 / distance);
    for (let j = 0; j <= n; j += 1) {
      const s = Math.min(distance * j, line.s13);
      const r = line.Position(s, geographiclib.Geodesic.LONG_UNROLL);
      coords.push([r.lon2, r.lat2]);
    }
  }
  return new OlGeomPolygon([coords]).transform(geographicProj, projection);
}

/**
 * Calculate area of a polygon with GeographicLib library
 * @param {*} polygon
 * @returns {Number} - area in square meters
 */
export function getGeographicLibArea(polygon) {
  const coordinates = polygon.getCoordinates()[0];
  if (coordinates.length < 3) return 0;
  const geoPoly = geod.Polygon(false);
  coordinates.forEach((coord) => {
    // flip lat/lon position
    geoPoly.AddPoint(coord[1], coord[0]);
  });
  const { area } = geoPoly.Compute(false, true);
  return Math.abs(area);
}

/**
 * Calculate distance of a line with GeographicLib library
 * @param {*} polygon
 * @returns {Number} - distance in meters
 */
export function getGeographicLibDistance(line) {
  let totalDistance = 0;
  line.forEachSegment((segStart, segEnd) => {
    const r = geod.Inverse(segStart[1], segStart[0], segEnd[1], segEnd[0]);
    totalDistance += r.s12;
  });
  return totalDistance;
}
