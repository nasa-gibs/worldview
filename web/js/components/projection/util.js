export function parse(state, errors, config) {
  // Permalink version 1.0 - 1.1
  if (state.switch) {
    state.p = state.switch;
    delete state.switch;
  }
  var projId = state.p;
  if (projId) {
    if (!config.projections[projId]) {
      delete state.p;
      errors.push({
        message: 'Unsupported projection: ' + projId
      });
    }
  }
}
