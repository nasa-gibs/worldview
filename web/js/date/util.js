import d3 from 'd3';

export function timelineDataHightlight(layerId, isActivate) {
  if (isActivate) {
    $('.data-bar-hovered').removeClass('data-bar-hovered');
    d3.select(
      '#timeline-footer svg g.plot rect[data-layer="' + layerId + '"]'
    ).classed('data-bar-hovered', true);
  } else {
    d3.select(
      '#timeline-footer svg g.plot rect[data-layer="' + layerId + '"]'
    ).classed('data-bar-hovered', false);
  }
}
export function getMaxTimelineWidth(ui) {
  var timeline = ui.timeline;
  var $footer = $('#timeline-footer');
  var $elWidth = $footer.width();
  var $dataWidth = timeline.x(timeline.data.end());
  if ($elWidth > $dataWidth) {
    return $dataWidth;
  }
  return $elWidth;
}
