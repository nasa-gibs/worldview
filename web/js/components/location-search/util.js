import CoordinateParser from 'coordinate-parser';
import util from '../../util/util';

/**
 * check for valid coordinates using https://www.npmjs.com/package/coordinate-parser
 * @param {String} position
 * @returns {Bool}
 */
export function isValidCoordinates(position) {
  try {
    const isValid = new CoordinateParser(position);
    return isValid;
  } catch (error) {
    return false;
  }
}

/**
 * Get fixed precision coordinate number
 * @param {String} coordinate
 * @returns {Number} parsed coordinate with fixed precision
 */
export const getCoordinateFixedPrecision = (coordinate) => Number(coordinate.toFixed(4));

/**
 * getFormattedCoordinates
 *
 * @param {String} latitude
 * @param {String} longitude
 *
 * @returns {Array} formattedCoordinates
 */
export const getFormattedCoordinates = (latitude, longitude) => {
  const parsedLatitude = getCoordinateFixedPrecision(latitude);
  const parsedLongitude = getCoordinateFixedPrecision(longitude);

  // format coordinates based on localStorage preference
  const format = util.getCoordinateFormat();
  const coordinates = util.formatCoordinate(
    [parsedLongitude, parsedLatitude],
    format,
  );
  const formattedCoordinates = coordinates.split(',');
  return formattedCoordinates;
};
