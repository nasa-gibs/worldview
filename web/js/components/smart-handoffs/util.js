import googleTagManager from 'googleTagManager';
import util from '../../util/util';

/**
 * Method call to direct the user to Earthdata Search with the necessary URL parameters that
 * encapsulate what the user is intending to try and download data / granules from
 * @param {*} proj
 * @param {*} includeDates
 * @param {*} selectedDate
 * @param {*} selectedCollection
 * @param {*} extentCoords
 * @param {*} showBoundingBox
 */
export default function openEarthDataSearch(proj, includeDates, selectedDate, selectedCollection, extentCoords, showBoundingBox) {
  const { value } = selectedCollection;
  googleTagManager.pushEvent({
    event: 'smart_handoffs_open_eds_data_query',
    selected_collection: value,
  });
  const PROJ_CODES = {
    arctic: '90!0!0!0!0!0',
    geographic: '0.0!-180.0!0!1!0!0,2',
    antarctic: '-90!180!0!2!0!0',
  };
  const { southWest, northEast } = extentCoords;
  const params = {
    q: value,
    p: value,
    m: PROJ_CODES[proj],
    'sb[0]': showBoundingBox ? `${southWest},${northEast}` : undefined,
  };

  if (includeDates) {
    const startDate = `${selectedDate}T00:00:00.000Z`;
    const endDate = `${selectedDate}T23:59:59.999Z`;
    params['[qt]'] = `${startDate},${endDate}`;
  }

  const earthDataSearchURL = `https://search.earthdata.nasa.gov/search/granules${util.toQueryString(params)}`;
  window.open(earthDataSearchURL, '_blank');
}
