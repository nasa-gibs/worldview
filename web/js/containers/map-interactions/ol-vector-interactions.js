import { connect } from 'react-redux';
import {
  debounce as lodashDebounce,
  get as lodashGet,
  includes as lodashIncludes,
  groupBy as lodashGroupBy,
} from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { transform } from 'ol/proj';
import { isFromActiveCompareRegion } from '../../modules/compare/util';
import { hasNonClickableVectorLayer } from '../../modules/layers/util';
import { getActiveLayers } from '../../modules/layers/selectors';
import vectorDialog from '../vector-dialog';
import { onMapClickGetVectorFeatures } from '../../modules/vector-styles/util';
import { openCustomContent, onClose } from '../../modules/modal/actions';
import { selectVectorFeatures as selectVectorFeaturesActionCreator } from '../../modules/vector-styles/actions';
import { changeCursor as changeCursorActionCreator } from '../../modules/map/actions';
import { ACTIVATE_VECTOR_ALERT } from '../../modules/alerts/constants';
import util from '../../util/util';

const { events } = util;

export class VectorInteractions extends React.Component {
  constructor(props) {
    super(props);
    this.mouseMove = lodashDebounce(this.mouseMove.bind(this), 8);
    this.singleClick = this.singleClick.bind(this);
  }

  componentDidMount() {
    events.on('map:mousemove', this.mouseMove);
    events.on('map:singleclick', this.singleClick);
  }

  componentWillUnmount() {
    events.off('map:mousemove', this.mouseMove);
    events.off('map:singleclick', this.singleClick);
  }

  mouseMove(event, map, crs) {
    const pixels = map.getEventPixel(event);
    const coord = map.getCoordinateFromPixel(pixels);

    const {
      isShowingClick, changeCursor, measureIsActive, compareState, swipeOffset, proj,
    } = this.props;
    const [lon, lat] = transform(coord, crs, 'EPSG:4326');
    if (lon < -250 || lon > 250 || lat < -90 || lat > 90) {
      return;
    }
    const hasFeatures = map.hasFeatureAtPixel(pixels);
    if (hasFeatures && !isShowingClick && !measureIsActive) {
      let isActiveLayer = false;
      map.forEachFeatureAtPixel(pixels, (feature, layer) => {
        const def = lodashGet(layer, 'wv.def');
        if (!def || lodashIncludes(def.clickDisabledFeatures, feature.getType())) return;
        const isWrapped = proj.id === 'geographic' && (def.wrapadjacentdays || def.wrapX);
        const isRenderedFeature = isWrapped ? lon > -250 || lon < 250 || lat > -90 || lat < 90 : true;
        if (isRenderedFeature && isFromActiveCompareRegion(pixels, layer.wv, compareState, swipeOffset)) {
          isActiveLayer = true;
        }
      });
      if (isActiveLayer) changeCursor(true);
    } else if (!hasFeatures && isShowingClick) {
      changeCursor(false);
    }
  }

  singleClick(e, map) {
    const {
      lastSelected, openVectorDialog, onCloseModal, selectVectorFeatures,
      modalState, getDialogObject, measureIsActive, isMobile, activeLayers,
      activateVectorAlert, proj,
    } = this.props;

    if (measureIsActive) return;
    const isVectorModalOpen = modalState.id.includes('vector_dialog') && modalState.isOpen;
    const pixels = e.pixel;
    const clickObj = getDialogObject(pixels, map);
    const metaArray = clickObj.metaArray || [];
    const selected = clickObj.selected || {};
    const offsetLeft = clickObj.offsetLeft || 10;
    const offsetTop = clickObj.offsetTop || 100;
    const isCoordinatesMarker = clickObj.isCoordinatesMarker || false;
    const dialogId = isVectorModalOpen ? modalState.id : `vector_dialog${pixels[0]}${pixels[1]}`;

    if (isCoordinatesMarker) {
      return;
    }

    if (metaArray.length) {
      openVectorDialog(dialogId, metaArray, offsetLeft, offsetTop, isMobile);
    } else {
      const mapRes = map.getView().getResolution();
      const hasNonClickableVectorLayerType = hasNonClickableVectorLayer(activeLayers, mapRes, proj.id);

      if (hasNonClickableVectorLayerType) {
        activateVectorAlert();
      }
    }
    if (Object.entries(selected).length || (Object.entries(lastSelected).length && !isVectorModalOpen)) {
      selectVectorFeatures(selected);
    } else if (isVectorModalOpen && !Object.entries(selected).length) {
      onCloseModal();
      selectVectorFeatures({});
    }
  }

  render() {
    return null;
  }
}

function mapStateToProps(state) {
  const {
    modal, map, measure, vectorStyles, browser, compare, proj, ui,
  } = state;
  let swipeOffset;
  const activeLayers = getActiveLayers(state);
  if (compare.active && compare.mode === 'swipe') {
    const percentOffset = state.compare.value || 50;
    swipeOffset = browser.screenWidth * (percentOffset / 100);
  }

  return {
    modalState: modal,
    isShowingClick: map.isClickable,
    isDistractionFreeModeActive: ui.isDistractionFreeModeActive,
    getDialogObject: (pixels, olMap) => onMapClickGetVectorFeatures(pixels, olMap, state, swipeOffset),
    lastSelected: vectorStyles.selected,
    measureIsActive: measure.isActive,
    isMobile: browser.lessThan.medium,
    compareState: compare,
    swipeOffset,
    proj,
    activeLayers,
  };
} const mapDispatchToProps = (dispatch) => ({
  selectVectorFeatures: (features) => {
    setTimeout(() => {
      dispatch(selectVectorFeaturesActionCreator(features));
    }, 1);
  },
  changeCursor: (bool) => {
    dispatch(changeCursorActionCreator(bool));
  },
  openCustomAlertModal: ({ id, props }) => {
    dispatch(openCustomContent(id, props));
  },
  onCloseModal: () => {
    dispatch(onClose());
  },
  activateVectorAlert: () => dispatch({ type: ACTIVATE_VECTOR_ALERT }),
  openVectorDialog: (dialogId, metaArray, offsetLeft, offsetTop, isMobile) => {
    const dialogKey = new Date().getUTCMilliseconds();
    dispatch(openCustomContent(dialogId,
      {
        backdrop: false,
        clickableBehindModal: true,
        desktopOnly: true,
        isDraggable: true,
        wrapClassName: 'vector-modal-wrap',
        modalClassName: 'vector-modal light',
        CompletelyCustomModal: vectorDialog,
        isResizable: true,
        dragHandle: '.modal-header',
        dialogKey,
        key: dialogKey,
        vectorMetaObject: lodashGroupBy(metaArray, 'id'),
        width: isMobile ? 250 : 445,
        height: 300,
        offsetLeft,
        offsetTop,
        timeout: 0,
        onClose: () => {
          setTimeout(() => {
            dispatch(selectVectorFeaturesActionCreator({}));
          }, 1);
        },
      }));
  },
});

VectorInteractions.propTypes = {
  changeCursor: PropTypes.func.isRequired,
  getDialogObject: PropTypes.func.isRequired,
  isShowingClick: PropTypes.bool.isRequired,
  measureIsActive: PropTypes.bool.isRequired,
  modalState: PropTypes.object.isRequired,
  onCloseModal: PropTypes.func.isRequired,
  openVectorDialog: PropTypes.func.isRequired,
  selectVectorFeatures: PropTypes.func.isRequired,
  compareState: PropTypes.object,
  isMobile: PropTypes.bool,
  lastSelected: PropTypes.object,
  proj: PropTypes.object,
  swipeOffset: PropTypes.number,
  activeLayers: PropTypes.array,
  activateVectorAlert: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(VectorInteractions);
