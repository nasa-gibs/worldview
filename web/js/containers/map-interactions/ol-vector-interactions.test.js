import React from 'react';
import renderer from 'react-test-renderer';
import util from '../../util/util';
import { VectorInteractions } from './ol-vector-interactions';
import { registerProjections } from '../../fixtures';

const { events } = util;
let component;
let map;
let changeCursor;
let selectVectorFeatures;
let getDialogObject;
let openVectorDialog;
beforeEach(() => {
  registerProjections();
  changeCursor = jest.fn();
  openVectorDialog = jest.fn();
  selectVectorFeatures = jest.fn();
  getDialogObject = () => ({
    metaArray: [0], selected: [1], offsetLeft: 100, offsetTop: 100,
  });
  component = renderer.create(
    <VectorInteractions
      mouseEvents={events}
      isShowingClick={false}
      changeCursor={changeCursor}
      getDialogObject={getDialogObject}
      openVectorDialog={openVectorDialog}
      selectVectorFeatures={selectVectorFeatures}
      lastSelected={{}}
      measureIsActive={false}
      onCloseModal={jest.fn()}
      modalState={{ id: [], isOpen: false }}
      isDistractionFreeModeActive={false}
    />,
  );
  map = {
    getEventPixel: jest.fn(),
    getCoordinateFromPixel: () => [0, 0],
    hasFeatureAtPixel: () => false,
  };
});

test('if there is a feature at pixel dispatch changeCursor action', () => {
  map.hasFeatureAtPixel = () => true;
  events.trigger('map:mousemove', {}, map, 'EPSG:3413');
  doAsync(() => expect(changeCursor.mock.calls.length).toBe(1));
});
test('if there is a feature at pixel on click get dialog', () => {
  events.trigger('map:singleclick', { pixel: [0, 0] }, map, 'EPSG:4326');
  expect(changeCursor.mock.calls.length).toBe(0);
  expect(selectVectorFeatures.mock.calls.length).toBe(1);
  expect(openVectorDialog.mock.calls.length).toBe(1);
});
test('if there is not a feature at pixel do not dispatch changeCursor action', () => {
  map.hasFeatureAtPixel = () => false;
  events.trigger('map:mousemove', { pixel: [0, 0] }, map, 'EPSG:4326');
  doAsync(() => expect(changeCursor.mock.calls.length).toBe(0));
});
test('Check that hover changes', () => {
  map.hasFeatureAtPixel = () => false;
  events.trigger('map:mousemove', { pixel: [0, 0] }, map, 'EPSG:4326');
  doAsync(() => expect(changeCursor.mock.calls.length).toBe(0));
});
test('Check that cursor-hover class is not present', () => {
  expect(component.toJSON()).toMatchSnapshot();
});
function doAsync(c) {
  setTimeout(() => {
    c(true);
  }, 10);
}
