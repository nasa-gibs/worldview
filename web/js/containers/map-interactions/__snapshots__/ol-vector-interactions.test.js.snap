// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Check that cursor-hover class is not present 1`] = `null`;
